arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedParquetDefinitionLevelReader.java

```diff
@@ -150,280 +159,195 @@ public void nextDictEncodedBatch(
-        FieldVector vector, int idx, ValuesAsBytesReader valuesReader, Mode mode);
+        FieldVector vector, int idx, VectorizedValuesReader valuesReader, int typeWidth);
 
     protected abstract void nextDictEncodedVal(
         FieldVector vector,
         int idx,
-        VectorizedDictionaryEncodedParquetValuesReader dictionaryEncodedValuesReader,
+        VectorizedDictionaryEncodedParquetValuesReader reader,
+        int numValuesToRead,
         Dictionary dict,
-        Mode mode,
+        NullabilityHolder nullabilityHolder,
+        int typeWidth,
+        Mode mode);
+
+    protected void nextValues(
+        FieldVector vector,
+        int bufferIdx,
         int numValues,
-        NullabilityHolder holder,
-        int typeWidth);
+        VectorizedValuesReader valuesReader,
+        int typeWidth) {
+      for (int i = 0; i < numValues; i++) {
+        nextVal(vector, bufferIdx + i, valuesReader, typeWidth);
+      }
+    }
+
+    // Used only for fixed-length binary types
+    @SuppressWarnings("checkstyle:VisibilityModifier")
+    protected byte[] byteArray;
   }
 
-  class LongReader extends NumericBaseReader {
+  class LongReader extends BaseReader {
     @Override
     protected void nextVal(
-        FieldVector vector, int idx, ValuesAsBytesReader valuesReader, Mode mode) {
-      vector.getDataBuffer().setLong(idx, valuesReader.readLong());
+        FieldVector vector, int idx, VectorizedValuesReader valuesReader, int typeWidth) {
+      vector.getDataBuffer().setLong((long) idx * typeWidth, valuesReader.readLong());
     }
 
     @Override
     protected void nextDictEncodedVal(
         FieldVector vector,
         int idx,
-        VectorizedDictionaryEncodedParquetValuesReader dictionaryEncodedValuesReader,
+        VectorizedDictionaryEncodedParquetValuesReader reader,
+        int numValuesToRead,
         Dictionary dict,
-        Mode mode,
-        int numValues,
-        NullabilityHolder holder,
-        int typeWidth) {
+        NullabilityHolder nullabilityHolder,
+        int typeWidth,
+        Mode mode) {
       if (Mode.RLE.equals(mode)) {
-        dictionaryEncodedValuesReader
+        reader
             .longDictEncodedReader()
-            .nextBatch(vector, idx, numValues, dict, holder, typeWidth);
+            .nextBatch(vector, idx, numValuesToRead, dict, nullabilityHolder, typeWidth);
       } else if (Mode.PACKED.equals(mode)) {
         vector
             .getDataBuffer()
-            .setLong(
-                (long) idx * typeWidth,
-                dict.decodeToLong(dictionaryEncodedValuesReader.readInteger()));
+            .setLong((long) idx * typeWidth, dict.decodeToLong(reader.readInteger()));
       }
     }
+
+    @Override
+    protected void nextValues(
+        FieldVector vector,
+        int bufferIdx,
+        int numValues,
+        VectorizedValuesReader valuesReader,
+        int typeWidth) {
+      valuesReader.readLongs(numValues, vector, bufferIdx);
```

Comment:
For example, `LongReader` overrides nextValues to use `VectorizedValuesReader.readLongs()` method for vectorized reading.